browser/mapping.go

```diff
@@ -774,14 +774,14 @@ func getOrInitBrowser(
 		vu.registerPid(pid)
 	}
 
-	vu.setBrowser(id, b)
-
 	go func(ctx context.Context) {
 		<-ctx.Done()
 		b.Close()
 		vu.deleteBrowser(id)
 	}(vu.Context())
 
+	vu.setBrowser(id, b)
```

Comment:
Just for curiosity. What is the reasoning for this change? Can we have an "race condition" between the `ctx.Done` and `vu.deleteBrowser` with the `vu.setBrowser` ?

Reply:
The reason for the change was to ensure that we still call `deleteBrowser` if a panic occurs in `setBrowser`. I know that should never happen. I just thought that the goroutine setup of the cleanup should be closer to where the browser process was initialised 🙂 There's no race conditions as far as i know.